### PR TITLE
[IMP] website: autocorrect website domain

### PR DESCRIPTION
**Pull request by JKE-be**
Before, get_http_domain make the job,
now we force website.domain to be valid

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/models/website.py [hunk 1]
```diff
@@ -224,11 +223,23 @@ def write(self, values):
...
     def _handle_favicon(self, vals):
         if 'favicon' in vals:
             vals['favicon'] = tools.image_process(vals['favicon'], size=(256, 256), crop='center', output_format='ICO')
 
+    @api.model
+    def _handle_domain(self, vals):
+        if 'domain' in vals and vals['domain']:
+            if not vals['domain'].startswith('http'):
+                vals['domain'] = 'https://%s' % vals['domain']
+            vals['domain'].rstrip('/')
```
Comment (rdeodoo) [inline addons/website/models/website.py:241]:
```suggestion
            vals['domain'] = vals['domain'].rstrip('/')
```

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Then should be ok, tests will come with PR from 14.2

**Comment by JKE-be** [comment]
> Before, get_http_domain make the job,

https://github.com/odoo/odoo/pull/71160

**Comment by JKE-be** [comment]
@robodoo r+

**Pull request closed without merging**